roles/cs.aws-autoscaling/defaults/main.yml

```diff
@@ -7,9 +7,49 @@ autoscaling_instance_tags: {}
 autoscaling_asg_min_size: 1
 autoscaling_asg_max_size: 1
 autoscaling_asg_desired_capacity: 1
-autoscaling_asg_wait_timeout: 1200
 autoscaling_asg_tags: {}
-autoscaling_asg_replace_all: yes
 autoscaling_asg_termination_policies: NewestInstance
-autoscaling_asg_wait_for_instances: yes
-autoscaling_asg_replace_with_check: no
\ No newline at end of file
+
+# The minimum time to wait since the node was started until it's considered
+# healthy, attached to the ASG (and next node can be replaced or refresh finished).
+# Note: In other words - it's the time for the node to warm itself up.
+autoscaling_healthcheck_grace_period: 30
+
+### Built-in AWS rolling ASG instance refresh (update / replacement)
+# The relative minimum amount of instances that must must be up and healthy
+# int the ASG at any point in time druing the refresh. (90% is the AWS default)
+#
+# Note: Max capacity does *not* have to be adjusted. AWS ignores it when fullfilling
+# the parameters defined here for the period of refresh.
+#
+# Couple of example cases:
+# - If we have one instance then the default 90% will ensure that this instance
+#   is not killed before the new one is up, healthy and after the grace period.
+# - If this is set to 0 the all old instances are terminated *at once* and new
+#   replacements started.
+# - Assuming that there are 3 instances in the group and this param is set to
+#   60%. The 60% out of 3 rounded up translates to 2, which means that
+#   at any point in time there will be *at least* 2 healthy instances in the ASG.
+#   Only one can be terminated / replaced at a time, so the full refresh will be
+#   made in 3 steps replacing them one-by-one.
+autoscaling_rolling_instance_refresh_min_healthy_percent: 90
```

Comment:
What if we have only one instance? It will add an instance or it will wait forever?
Taking into account that we want to do a setup upgrade - isn't that better to drop 2 and leave just one? 

3 instances are rare among our customers. I would rather say 50%

Reply:
This is up for discussion, I've added a safe default (the same as in aws). We can override it anytime.

Reply:
It will not wait forever of course, it will add an extra one.